File: Basic/G.F Series/g.f-series.py
class Solution:
    def gfSeries(self, N : int) -> None:
        fib1 = 0
        fib2 = 1
        fib = 0
        if(N == 1):
            print(fib1)
        if(N == 2):
            print(fib1,fib2)
        else:
            print(fib1,fib2,end = " ")
            for i in range(2,N):
                fib = fib1**2 - fib2
                print(fib,end = " ")
                fib1 = fib2
                fib2 = fib
        print()



#{ 
 # Driver Code Starts
if __name__=="__main__":
    t = int(input())
    for _ in range(t):
        
        N = int(input())
        
        obj = Solution()
        obj.gfSeries(N)

# } Driver Code Ends

File: Basic/G.F Series/g.f-series.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++
class Solution
{
public:
    void gfSeries(int N)
    {
        // Write Your Code here
        long long a = 0;
        long long b = 1;
        N -= 2;
        cout << a << " " << b << " ";
        while(N--){
            long long c = (a*a)-b;
            a = b;
            b = c;
            cout<<c<<" ";
        }
        cout << endl;
        return;
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int N;
        cin>>N;
        Solution ob;
        ob.gfSeries(N);
    }
    return 0;
}
// } Driver Code Ends